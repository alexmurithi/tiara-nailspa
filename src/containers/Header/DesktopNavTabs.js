import React, { useState } from 'react'
import { Box, Tabs, Tab } from '@mui/material'

import { navigationTabs } from 'src/utils/Navigation'
import desktopNavTabsStyles from '@styles/containers/DesktopNavTabsStyles'

const DesktopNavTabs = () => {
  const { classes } = desktopNavTabsStyles()
  const [tabValue, setDesktopTabValue] = useState(0)

  const handleDesktopTabs = (event, newValue) => {
    setDesktopTabValue(newValue)
  }

  return (
    <Box className={classes.root}>
      <Tabs
        textColor="primary"
        value={tabValue}
        onChange={handleDesktopTabs}
        centered
        className={classes.desktopTabs}
      >
        {navigationTabs.map((tab, index) => (
          <Tab
            key={index}
            label={tab}
            disableRipple
            className={classes.desktopTab}
          />
        ))}
      </Tabs>
    </Box>
  )
}

export default React.memo(DesktopNavTabs)
