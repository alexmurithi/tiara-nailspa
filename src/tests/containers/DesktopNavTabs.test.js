/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import renderer from 'react-test-renderer'
import DesktopNavTabs from '@containers/Header/DesktopNavTabs'
import { ThemeProvider } from '@mui/material'
import theme from '@themes/Default'

describe('<DesktopNavTabs/>', () => {
  it('renders correctly without crashing', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <DesktopNavTabs />
        </ThemeProvider>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
