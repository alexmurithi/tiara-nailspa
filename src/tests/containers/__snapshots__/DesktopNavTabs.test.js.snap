// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DesktopNavTabs/> renders correctly without crashing 1`] = `
<div
  className="MuiBox-root css-gs389c-root"
>
  <div
    className="MuiTabs-root css-s7ydoq-MuiTabs-root-desktopTabs"
  >
    <div
      className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
      style={
        Object {
          "marginBottom": -0,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-orientation={null}
        className="MuiTabs-flexContainer MuiTabs-centered css-1wf8b0h-MuiTabs-flexContainer"
        onKeyDown={[Function]}
        role="tablist"
      >
        <button
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1423l1g-MuiButtonBase-root-MuiTab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          home
          <span
            className="MuiTabs-indicator css-u83qgz-MuiTabs-indicator"
            style={Object {}}
          />
        </button>
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1423l1g-MuiButtonBase-root-MuiTab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          services
        </button>
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1423l1g-MuiButtonBase-root-MuiTab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          pricing
        </button>
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1423l1g-MuiButtonBase-root-MuiTab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          blog
        </button>
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1423l1g-MuiButtonBase-root-MuiTab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          gallery
        </button>
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1423l1g-MuiButtonBase-root-MuiTab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          shop
        </button>
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1423l1g-MuiButtonBase-root-MuiTab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          about
        </button>
      </div>
    </div>
  </div>
</div>
`;
